### PR TITLE
x64: add x64 encoding benchmarks

### PR DESCRIPTION
**Pull request by abrown**
This is a follow-up to #2819 to add a benchmark comparing the different encoding approaches (builder vs function) used to encode EVEX instructions. The measurements I observed indicate that the builder pattern is faster, as noted in #2819. The more interesting improvement of this PR is to show how to measure parts of the `cranelift-codegen` crate. To do so with `criterion`, I had to make some of the x64 module public:
 - the first commit moves the `encoding` module out of `x64::inst` so that `x64::inst` does not have to become public
 - the second makes the `x64` module itself public
 - the third adds the benchmark.

### CODE REVIEW

File: cranelift/codegen/benches/x64-evex-encoding.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+use cranelift_codegen::isa::x64::encoding::{
```
Comment (abrown) [inline cranelift/codegen/benches/x64-evex-encoding.rs:1]: I think this entire benchmark should be feature-gated (i.e., `#[cfg(feature = "x86")]`) but I don't know the best way to do this?

Comment (cfallin) [reply cranelift/codegen/benches/x64-evex-encoding.rs:1]: You could put the whole implementation inside an inner `mod { ... }` with the #[cfg(...)]`, export one main function, and have an alternate (empty) main function with `[cfg(not(feature = "x86"))]` probably?

**Review by cfallin — APPROVED** [review]
Looks good! If you want to look into it, I think there's a way to feature-gate the whole bench on x86 being compiled in, but since it's not built by default (needs a specific `cargo bench` invocation) I think it's not the end of the world if that doesn't work either.

File: cranelift/codegen/benches/x64-evex-encoding.rs (hunk 1 above)
Comment (abrown) [reply cranelift/codegen/benches/x64-evex-encoding.rs]: Ok, I finally figured out how to do this. My latest update uses the inner module approach feature-gated on `x86` and prints a "can't run this benchmark" message if the feature is not present.

**Review by cfallin — APPROVED** [review]
Looks good!

**Pull request merged**